src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -234,12 +225,118 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
+            //
+            // This approach would ensure the computation of #2 will not be the cause of such delays, with the obvious trade off of potentially not providing
+            // expanded items until later (or never) in a completion session even if the feature is enabled. Note that in most cases we'd expect task #2 to finish
+            // in time and complete result would be available from the start of the session. However, even in the case only partial result is returned at the start,
+            // we still believe this is acceptable given how critical perf is in typing scenario.
+            // Additionally, expanded items are usually considered complementary. The need for them only rise occasionally (it's rare when users need to add imports,)
+            // and when they are needed, our hypothesis is because of their more intrusive nature (adding an import to the document) users would more likely to
+            // contemplate such action thus typing slower before commit and/or spending more time examining the list, which give us some opportunities
+            // to still provide those items later before they are truly required.            
+
+            if (!options.ShouldShowItemsFromUnimportNamspaces())
+            {
+                // No need to trigger expanded providers at all if the feature is disabled, just trigger core providers and return;
+                var (context, list) = await GetCompletionContextWorkerAsync(document, trigger, triggerLocation,
+                    options with { ExpandedCompletionBehavior = ExpandedCompletionMode.NonExpandedItemsOnly }, cancellationToken).ConfigureAwait(false);
+                AddPropertiesToSession(session, list, triggerLocation);
+                return context;
+            }
+
+            if (!session.TextView.Options.GetOptionValue(DefaultOptions.ResponsiveCompletionOptionId))
+            {
+                // We tie the behavior of delaying expand items to editor's "responsive completion" option.
+                // i.e. "responsive completion" disabled == always wait for all items to be calculated.
+                options = options with
+                {
+                    BlockOnExpandedCompletion = true,
+                    ExpandedCompletionBehavior = ExpandedCompletionMode.AllItems,
+                };
+                var (context, list) = await GetCompletionContextWorkerAsync(document, trigger, triggerLocation, options, cancellationToken).ConfigureAwait(false);
+                AddPropertiesToSession(session, list, triggerLocation);
+                return context;
+            }
+
+            var expandedItemsTask = Task.Run(async () => await GetCompletionContextWorkerAsync(document, trigger, triggerLocation,
+                options with { ExpandedCompletionBehavior = ExpandedCompletionMode.ExpandedItemsOnly }, cancellationToken).ConfigureAwait(false),
+                cancellationToken);
```

Comment:
should we set BlockOnExpandedCompletion explicitly to false here?

Reply:
BlockOnExpandedCompletion  is no longer needed actually, so I removed it